crates/autopilot/src/solvable_orders.rs

```diff
@@ -226,6 +238,63 @@ impl SolvableOrdersCache {
     pub fn last_update_time(&self) -> Instant {
         self.cache.lock().unwrap().orders.update_time
     }
+
+    /// Quotes all limit orders and sets the fee_amount for each one to the fee returned by the
+    /// quoting process. If quoting fails, the corresponding order is filtered out.
+    async fn set_limit_order_fees(&self, orders: Vec<Order>) -> Vec<Order> {
+        stream::iter(orders.into_iter())
+            .filter_map(|mut order| async {
+                if order.metadata.class != OrderClass::Limit {
+                    return Some(order);
+                }
+
+                match self
+                    .quoter
+                    .calculate_quote(QuoteParameters {
+                        sell_token: order.data.sell_token,
+                        buy_token: order.data.buy_token,
+                        side: match order.data.kind {
+                            OrderKind::Buy => OrderQuoteSide::Buy {
+                                buy_amount_after_fee: order.data.buy_amount,
+                            },
+                            OrderKind::Sell => OrderQuoteSide::Sell {
+                                sell_amount: SellAmount::AfterFee {
+                                    value: order.data.sell_amount,
+                                },
+                            },
+                        },
+                        from: order.metadata.owner,
+                        app_data: order.data.app_data,
+                        signing_scheme: match order.signature {
+                            Signature::Eip712(_) => QuoteSigningScheme::Eip712,
+                            Signature::EthSign(_) => QuoteSigningScheme::EthSign,
+                            Signature::Eip1271(_) => QuoteSigningScheme::Eip1271 {
+                                onchain_order: false,
+                                verification_gas_limit: default_verification_gas_limit(),
+                            },
+                            Signature::PreSign => QuoteSigningScheme::PreSign {
+                                onchain_order: false,
+                            },
+                        },
+                    })
+                    .await
+                {
+                    Ok(quote) => {
+                        order.data.fee_amount = quote.fee_amount;
```

Comment:
I talked to Nick and I advised you wrong. The idea was indeed to use a new field for this fee amount instead of overwriting the existing `fee` amount.

Reply:
I see. Why?

Reply:
1) Adding a new field would allow us to have an additional flat `fee` amount not equal to 0 in the future.
2) The `fee` amount is a value that gets signed by the user when creating the order so modifying it afterwards could make things hairy eventually.

Reply:
https://github.com/cowprotocol/services/pull/690/commits/3a62da0c469cf63b96c653792e1b14c313c5afa7